plugin/mlsdbprovider.cpp

```diff
@@ -287,10 +290,19 @@ void MlsdbProvider::timerEvent(QTimerEvent *event)
         m_fixLostTimer.stop();
         setStatus(StatusAcquiring);
     } else if (event->timerId() == m_recalculatePositionTimer.timerId()) {
-        if (m_positioningEnabled && (m_signalUpdateCell || m_signalUpdateWlan)) {
-                m_signalUpdateCell = false;
-                m_signalUpdateWlan = false;
-                calculatePositionAndEmitLocation();
+        const qint64 currTimestamp = QDateTime::currentMSecsSinceEpoch();
+        if (!m_positioningEnabled) {
+            qCDebug(lcGeoclueMlsdb) << "positioning is disabled, preventing MLS calculation";
+        } else if (m_currentLocation.timestamp() == 0
+                || ((currTimestamp - m_currentLocation.timestamp()) > ReuseInterval)
```

Comment:
I was pondering if the backof should be in the m_idleTimer directly, then we would not do extra calculations, right? Now if ther is no cell or wlan updates it runs a loop extra round?

Reply:
Do you mean the m_recalculatePositionTimer?  One reason to keep it the current way is that we will still use cell-id based positioning to calculate an update even if we're throttled for a long period of time (e.g. 6 or 10 minutes).  Even though the accuracy of locally-calculated cell-id fixes are very low, it may be more accurate in reality than a 6-minute-old update.